common/src/main/java/org/apache/uniffle/common/netty/protocol/SendShuffleDataRequest.java

```diff
@@ -103,7 +103,19 @@ private static Map<Integer, List<ShuffleBlockInfo>> decodePartitionData(ByteBuf
       int lengthOfShuffleBlocks = byteBuf.readInt();
       List<ShuffleBlockInfo> shuffleBlockInfoList = Lists.newArrayList();
       for (int j = 0; j < lengthOfShuffleBlocks; j++) {
-        shuffleBlockInfoList.add(Decoders.decodeShuffleBlockInfo(byteBuf));
+        try {
+          shuffleBlockInfoList.add(Decoders.decodeShuffleBlockInfo(byteBuf));
+        } catch (Throwable t) {
+          // An OutOfDirectMemoryError will be thrown, when the direct memory reaches the limit.
+          // OutOfDirectMemoryError does not cause the jvm exit and cause the direct memory leak.
```

Comment:
OutOfDirectMemoryError does not cause the JVM to exit and it may lead to direct memory leaks.

Reply:
Fixed.
@rickyma